frame/child-bounties/src/lib.rs

```diff
@@ -0,0 +1,892 @@
+		ChildBountyAdded { index: BountyIndex, child_index: BountyIndex },
+		/// A child-bounty is awarded to a beneficiary.
+		ChildBountyAwarded {
+			index: BountyIndex,
+			child_index: BountyIndex,
+			beneficiary: T::AccountId,
+		},
+		/// A child-bounty is claimed by beneficiary.
+		ChildBountyClaimed {
+			index: BountyIndex,
+			child_index: BountyIndex,
+			payout: BalanceOf<T>,
+			beneficiary: T::AccountId,
+		},
+		/// A child-bounty is cancelled.
+		ChildBountyCanceled { index: BountyIndex, child_index: BountyIndex },
+	}
+
+	/// Number of total child bounties.
+	#[pallet::storage]
+	#[pallet::getter(fn child_bounty_count)]
+	pub type ChildBountyCount<T: Config> = StorageValue<_, BountyIndex, ValueQuery>;
+
+	/// Number of child-bounties per parent bounty.
+	#[pallet::storage]
+	#[pallet::getter(fn parent_child_bounties)]
+	pub type ParentChildBounties<T: Config> =
+		StorageMap<_, Twox64Concat, BountyIndex, BountyIndex, ValueQuery>;
+
+	/// Child-bounties that have been added.
+	#[pallet::storage]
+	#[pallet::getter(fn child_bounties)]
+	pub type ChildBounties<T: Config> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		BountyIndex,
+		Twox64Concat,
+		BountyIndex,
+		ChildBounty<T::AccountId, BalanceOf<T>, T::BlockNumber>,
+	>;
+
+	/// The description of each child-bounty.
+	#[pallet::storage]
+	#[pallet::getter(fn child_bounty_descriptions)]
+	pub type ChildBountyDescriptions<T: Config> =
+		StorageMap<_, Twox64Concat, BountyIndex, BoundedVec<u8, T::MaximumReasonLength>>;
+
+	/// The cumulative child-bounty curator fee for each parent bounty.
+	#[pallet::storage]
+	#[pallet::getter(fn children_curator_fees)]
+	pub type ChildrenCuratorFees<T: Config> =
+		StorageMap<_, Twox64Concat, BountyIndex, BalanceOf<T>, ValueQuery>;
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Add a new child-bounty.
+		///
+		/// The dispatch origin for this call must be master curator.
+		/// parent bounty must be in "active" state.
+		///
+		/// Child-bounty gets added successfully & fund gets transferred from
+		/// parent bounty to child-bounty account, if parent bounty has
+		/// enough fund. Else call gets failed.
+		///
+		/// Upper bound to maximum number of active  child-bounties that
+		/// can be added are managed via runtime trait config
+		/// 'MaxActiveChildBountyCount'.
+		///
+		/// If the call is success, the state of child-bounty is
+		/// moved to "Added" state.
+		///
+		/// - `bounty_id`: Bounty ID for which child-bounty to be added.
+		/// - `value`: Value for executing the proposal.
+		/// - `description`: Text description for the child-bounty.
+		#[pallet::weight(<T as Config>::WeightInfo::add_child_bounty(description.len() as u32))]
+		pub fn add_child_bounty(
+			origin: OriginFor<T>,
+			#[pallet::compact] parent_bounty_id: BountyIndex,
+			#[pallet::compact] value: BalanceOf<T>,
+			description: Vec<u8>,
+		) -> DispatchResult {
+			let signer = ensure_signed(origin)?;
+
+			// Verify the arguments.
```

Comment:
would be nice if checks that are done by the parent pallet are all abstracted away in on `fn verify_bounty(...) -> DispatchResult` in the parent pallet that gets reuse here.

Reply:
We do have a function that I am reusing in the child pallet - https://github.com/paritytech/substrate/blob/gd-child-bounties/frame/child-bounties/src/lib.rs#L781
But this is only relevant in the child pallet.